include/GoogleSync/GoogleSync.php

```diff
@@ -972,27 +1042,75 @@ public function setTimezone($timezone)
      *
      * @return bool Success/Failure
      */
-    protected function setLastSync($event_local, $gEventId = null)
+    protected function setLastSync(Meeting $event_local, $gEventId = null)
     {
-
         if (isset($gEventId)) {
             $event_local->gsync_id = $gEventId;
         }
 
         $event_local->gsync_lastsync = time() + 3; // we add three seconds to this, so that the modified time is always older than the gsync_lastsync time
         $return = $event_local->save(false);
 
-        // Set the meeting as accepted by the user, otherwise it doesn't show up on the calendar. We do it here because it must be saved first.
-        $event_local->set_accept_status($this->workingUser, 'accept');
-
-        if (isset($return)) {
+        $isValidator = new SuiteValidator();
+        if ($isValidator->isValidId($return)) {
+            $event_local->set_accept_status($this->workingUser, 'accept');  // Set the meeting as accepted by the user, otherwise it doesn't show up on the calendar. We do it here because it must be saved first.
             $this->syncedList[] = $event_local->id;
             return true;
         } else {
+            $this->logger->fatal(__FILE__ . ':' . __LINE__ . ' ' . __METHOD__ . ' - ' . 'Something went wrong saving the local record.');
             return false;
         }
     }
 
+    /**
+     * Helper function for doSync
+     * 
+     * @param Meeting $meeting The CRM Meeting
+     * @param \Google_Service_Calendar_Event $event The Google Event
+     * @param string $action The action to take with the two events
+     * 
+     * @return bool Success/Failure
+     */
+    protected function doAction(Meeting $meeting = null, Google_Service_Calendar_Event $event = null, $action)
```

Comment:
Wrong order of arguments. Last argument can not be requered if the previous arguments are optional. (also the function complexity is too large.)

Reply:
re-ordered arguments. I'll work on the complexity. The mess at the top is only used for logging. :|

Reply:
Fixed